src/generators/hl2c.ml

```diff
@@ -56,29 +56,34 @@ and function_entry = {
 	mutable fe_calling : function_entry list;
 }
 
-type context = {
+type global_context = {
 	version : int;
-	out : Buffer.t;
-	mutable tabs : string;
+	hash_mutex : Mutex.t;
 	hash_cache : (int, int32) Hashtbl.t;
 	hash_mem : (int32, bool) Hashtbl.t;
 	mutable hash_cache_list : int list;
 	hlcode : code;
 	dir : string;
-	mutable curfile : string;
 	mutable cfiles : string list;
 	ftable : function_entry array;
 	htypes : (ttype, string) PMap.t;
 	gnames : string array;
 	bytes_names : string array;
+	mutable type_module : (ttype, code_module) PMap.t;
+	gcon : Gctx.t;
+}
+
+type file_context = {
+	out : Buffer.t;
+	mutable tabs : string;
+	dir : string;
+	curfile : string;
 	mutable defines : string list;
 	defined_funs : (int, unit) Hashtbl.t;
 	hdefines : (string, unit) Hashtbl.t;
 	mutable defined_types : (ttype, unit) PMap.t;
-	mutable file_prefix : string;
 	mutable fun_index : int;
-	mutable type_module : (ttype, code_module) PMap.t;
-	gcon : Gctx.t;
```

Comment:
Hmm, this shouldn't be necessary because gctx is pretty const.

Reply:
Ah never mind, I thought you were passing this context around further below, but that's actually a different context also named gctx.